# AIM
This repo contains links to resources associated with AI/ML methods for medical images and videos. ??? Add reference here ???

OPEN AI MODELS:
HistoQC is an open-source quality control tool for digital pathology slides
HTTPS://GITHUB.COM/CHOOSEHAPPY/HISTOQC 

Towards a general-purpose foundation model for computational pathology
UNI, HIPT HTTPS://DOI.ORG/10.1038/S41591-024-02857-3 

CellViT++: Energy-Efficient and Adaptive Cell Segmentation and Classification Using Foundation
CELLVIT HTTPS://GITHUB.COM/TIO-IKIM/CELLVIT-PLUS-PLUS

	CELL POSE HTTPS://GITHUB.COM/MOUSELAND/CELLPOSE 
	PROV-GIGAPATH HTTPS://DOI.ORG/10.1038/S41586-024-07441-W , HTTPS://GITHUB.COM/PROV-GIGAPATH/PROV-GIGAPATH
	H0-MINI: HTTPS://HUGGINGFACE.CO/BIOPTIMUS/H0-MINI
	CONCH (VLM): HTTPS://GITHUB.COM/MAHMOODLAB/CONCH/TREE/MAIN
	GITHUB: HTTPS://GITHUB.COM/LEIMAO/DEEPLAB-V3, HTTPS://ARXIV.ORG/ABS/1706.05587
	CLAM: HTTPS://GITHUB.COM/MAHMOODLAB/CLAM
	GITHUB: HTTPS://GITHUB.COM/LUCIDRAINS/VIT-PYTORCH  PAPER: DOSOVITSKIY A, BEYER L, KOLESNIKOV A, WEISSENBORN D, ZHAI X, UNTERTHINER T, DEHGHANI M, MINDERER M, HEIGOLD G, GELLY S, USZKOREIT J. AN IMAGE IS WORTH 16X16 WORDS: TRANSFORMERS FOR IMAGE RECOGNITION AT SCALE. ARXIV PREPRINT ARXIV:2010.11929. 2020 OCT 22. LINK: HTTPS://ARXIV.ORG/ABS/2010.11929

HistomicsTK is a Python package for the analysis of digital pathology images.
HTTPS://GITHUB.COM/DIGITALSLIDEARCHIVE/HISTOMICSTK 

Slideflow is a deep learning library for digital pathology, offering a user-friendly interface for model development.
HTTPS://GITHUB.COM/SLIDEFLOW/SLIDEFLOW?TAB=README-OV-FILE 

A Pathology Foundation Model for Cancer Diagnosis and Prognosis Prediction
HTTPS://GITHUB.COM/HMS-DBMI/CHIEF 

Mask2Former: Masked-attention Mask Transformer for Universal Image Segmentation (CVPR 2022)
HTTPS://GITHUB.COM/FACEBOOKRESEARCH/MASK2FORMER

Detectron2 is Facebook AI Research's next generation library that provides state-of-the-art detection and segmentation algorithms.
HTTPS://GITHUB.COM/FACEBOOKRESEARCH/DETECTRON2


SAM (Segment Anything Model, META, 2023) 
HTTPS://GITHUB.COM/FACEBOOKRESEARCH/SEGMENT-ANYTHING

UNI: Towards a General-Purpose Foundation Model for Computational Pathology
HTTPS://GITHUB.COM/MAHMOODLAB/UNI

Sarder Lab: Codes for computational pathology from Pinaki Sarder's lab
https://github.com/SarderLab

	MOPADI - MORPHING HISTOPATHOLOGY DIFFUSION HTTPS://GITHUB.COM/KATHERLAB/MOPADI 

	OPEN DATASETS:
	KPMP HTTPS://WWW.KPMP.ORG/AVAILABLE-DATA
	HUBMAP HTTPS://PORTAL.HUBMAPCONSORTIUM.ORG/?_GL=1*C1FDYL*_GA*MTG4MTA1ODMWMS4XNZQWMJCZNDQ5*_GA_N77K0HBGRV*CZE3NTGWODUYMJQKBZIWJGCWJHQXNZU4MDG1MJI0JGO2MCRSMCROMA..
	TCGA HTTPS://WWW.CANCER.GOV/CCG/RESEARCH/GENOME-SEQUENCING/TCGA 
	GTEX HTTPS://GTEXPORTAL.ORG/HOME/ 
	NCI'S CPTAC (PROTEOMICS, GENOMICS, HISTOPATHOLOGY) HTTPS://GDC.CANCER.GOV/ABOUT-GDC/CONTRIBUTED-GENOMIC-DATA-CANCER-RESEARCH/CLINICAL-PROTEOMIC-TUMOR-ANALYSIS-CONSORTIUM-CPTAC 
	NCI'S IMAGING DATA COMMONS HTTPS://DATACOMMONS.CANCER.GOV/REPOSITORY/IMAGING-DATA-COMMONS 
	CZI'S HUMAN CELL ATLAS HTTPS://CELLXGENE.CZISCIENCE.COM/
	HTTPS://WWW.HUMANCELLATLAS.ORG/ 
	NCI'S HUMAN TUMOR ATLAS NETWORK HTTPS://HUMANTUMORATLAS.ORG/ 
	BRIDGE2AI'S FUNCTIONAL GENOMICS PROJECT (PROTEIN-PROTEIN INTERACTIONS, SINGLE-CELL IMAGING; TREY IDEKER AT UCSD IS PI) HTTPS://BRIDGE2AI.ORG/CM4AI_/ 
	4 DIMENSIONAL NUCLEOME (IMAGING AND OMICS TO RELATE THE SPATIAL ORIENTATION OF THE NUCLEOME TO GENE REGULATION) HTTPS://4DNUCLEOME.ORG/ 
	APOLLO PROGRAM (DOD, VA, NCI; CANCER PROGRAM WITH GENOMICS, PROTEOMICS, PATHOLOGY AND EXCELLENT LONGITUDINAL CLINICAL DATA OF VETERANS) HTTPS://DCTD.CANCER.GOV/PROGRAMS/OCCPR 
	HTTPS://CELLXGENE.CZISCIENCE.COM/DATASETS
	HTTPS://WWW.NCBI.NLM.NIH.GOV/GDS/ 
 























maybe include Grad-CAM updates/versions such as Grad-CAM++, EighenGrad-CAM etc - some references are included at the end of this GitHub
https://github.com/jacobgil/pytorch-grad-cam
https://arxiv.org/abs/2011.08891
Use HiResCAM instead of Grad-CAM for faithful explanations of convolutional neural networks Rachel L. Draelos, Lawrence Carin
https://arxiv.org/abs/1710.11063
Grad-CAM++: Improved Visual Explanations for Deep Convolutional Networks Aditya Chattopadhyay, Anirban Sarkar, Prantik Howlader, Vineeth N Balasubramanian
https://arxiv.org/abs/1910.01279
Score-CAM: Score-Weighted Visual Explanations for Convolutional Neural Networks Haofan Wang, Zifan Wang, Mengnan Du, Fan Yang, Zijian Zhang, Sirui Ding, Piotr Mardziel, Xia Hu
https://ieeexplore.ieee.org/abstract/document/9093360/
Ablation-cam: Visual explanations for deep convolutional network via gradient-free localization. Saurabh Desai and Harish G Ramaswamy. In WACV, pages 972–980, 2020
https://arxiv.org/abs/2008.02312
Axiom-based Grad-CAM: Towards Accurate Visualization and Explanation of CNNs Ruigang Fu, Qingyong Hu, Xiaohu Dong, Yulan Guo, Yinghui Gao, Biao Li
https://arxiv.org/abs/2008.00299
Eigen-CAM: Class Activation Map using Principal Components Mohammed Bany Muhammad, Mohammed Yeasin
http://mftp.mmcheng.net/Papers/21TIP_LayerCAM.pdf

<img width="518" height="690" alt="image" src="https://github.com/user-attachments/assets/c0d5f1a1-5e16-4f99-8503-763b519e2ce3" />
